### PR TITLE
Slider fixes (dynamic updating, no more jQuery UI)

### LINKED ISSUE #1581: Toggling slider for a domain on options page resets scroll
Causing you to lose your place. From a support email:

>When an entry is flipped from green to yellow or red it should not (yet) move and the display of the URLs not jump back to the beginning because this makes it very difficult to go through the whole list of maybe more than 1000 entries.

### LINKED ISSUE #1746: Domain slider tooltip only appears when hovering over bottom section of slider
On the Oct 25 release, if you hover your mouse over the sliders, the tooltip does not appear unless you move your mouse slightly below the slider itself (by changing your pointer from the clicker/4 directional arrows to the regular arrow).

### LINKED ISSUE #1863: Changing blocking preferences in tracking options sometimes resets unrelated origins
On the Tracking Domains options tab, after making several changes to closely-displayed but unrelated domains, a change to one domain sometimes triggers one of the others to get reset to its most recent position.  

release1
release2
release3
If I set virgin.15gifts.com to cookieblock, and then set 1rx.io to block, virgin.15gifts.com goes back to allowed.

If I set bat.bing.com to allow, then set bounceexchange.com to cookieblock, bat.bing.com goes back to cookieblock.

I have not seen this happen with origins that have not had their user preferences changed.

I cannot catch it reliably by hand, but I have this test case that always triggers it: I found this issue while writing tests for a somewhat related issue.
https://gist.github.com/mijiturka/989dc5958f62c8b2628efc1bf19fcc23

I managed to catch it a few times on the master branch in firefox-dev 59.0 and on the latest release in firefox 58.0.1. Have not tested in Chromium.

### LINKED ISSUE #1912: Options menu sliders do not register when clicking at the bottom of the slider
When manipulating sliders in the options menu, the sliders respond differently depending on where you click. If you click or click and drag in the middle of the slider, the mouse cursor turns into a hand, and the slider functions as you would expect. 
functioning sliders

If you click near the very bottom of the slider, the UI itself will react, but changes to tracker settings do not persist.
broken sliders

This is not the case with the popup sliders; the cursor does switch between a hand and a cursor depending on where in the slider it is, but the slider functions either way.

### PR DESCRIPTION
**Pull request by ghostwords**
- [x] Fixed #1746, fixed #1912 by removing jQuery UI sliders. Domain sliders were a [mish](https://github.com/EFForg/privacybadger/commit/6a3f8ea71a8c3175b69e830f0c6988e66590fcff)[mash](https://github.com/EFForg/privacybadger/commit/a602427119f0a36bb60108393f6e95137e9c5d91) of jQuery UI sliders and [CSS Toggle Switch](https://github.com/ghinda/css-toggle-switch).

    This also improves keyboard accessibility.

    The downside is that domain sliders are no longer draggable.

- [x] Fixes #1581, fixes #1863: The contents of the Tracking Domains tab are no longer rerendered in response to moving sliders, reverting control and removing domains. Instead, any sliders that should get updated or removed get updated or removed leaving the rest of list alone.

    For example, moving the slider for `example.com` should also move the sliders for `www.example.com` and `cdn.example.com` (assuming neither subdomain is already user-set).

   The popup isn't getting this feature for now (just like the popup currently doesn't rerender the domains list).

I attempted a fix for #2422 (434e3a2587d05ee6baa58ccfb32058d35f66d9e0) but rolled it back as the first attempt wasn't good enough (need to figure out how to cancel in-progress rendering in response to anything that triggers a rerender, such as updating a domains list filter), and I don't want it to hold up the other fixes.

What are the remaining uses of jQuery UI? Tabs, buttons, icons, anything else? Would be good to eventually remove jQuery UI entirely.

### CODE REVIEW

**Comment by ghostwords** [comment]
Removing jQuery UI sliders may also allow us to restore tooltips to Firefox (#1814). However, I haven't been able to reproduce the original problem (https://github.com/EFForg/privacybadger/issues/1763#issuecomment-342645171) using an old version of Firefox (52) with an old version of Privacy Badger (with fancy tooltips enabled). Until we figure out how to reproduce the original problem, we won't be able to say it's fixed with any confidence.

**Comment by ghostwords** [comment]
A good followup to this PR would be to continue the work in deaec2a1c20b1632601451133e124afb953d1222 and make keyboard navigation more intuitive by rearranging the order of elements in each slider row (and fixing the way they are laid out, maybe by replacing floats with ... flex?). We should then also fix keyboard navigation in the popup to correctly ignore hidden elements (error reporting form and share form elements are included in keyboard navigation regardless of their visibility).

Fixing slider row element order will then make it easier to add RTL locale support to the domains list (currently hardcoded to LTR).

File: src/tests/tests/htmlutils.js [hunk 1]
```diff
@@ -72,32 +47,33 @@ QUnit.test("getActionDescription", (assert) => {
...
 
   // Run each test.
-  for (var i = 0; i < tests.length; i++) {
-    var origin = tests[i].origin;
-    var action = tests[i].action;
-    var expected = tests[i].expectedResult;
-    var message = "Inputs: '" + origin + "' and '" + action + "'";
-    var html = htmlUtils.getToggleHtml(origin, action);
-    var inputValue = $('input[name="' + origin + '"]:checked', html).val();
+  for (let i = 0; i < tests.length; i++) {
```
Comment (ablanathtanalba) [inline src/tests/tests/htmlutils.js:71]: AFAIK there isn't a reason to use old school ES5 for loops anymore. A `forEach` or `for ... of` ought to work the same here, as well as making it more readable

**Comment by ablanathtanalba** [comment]
Finished doing some more manual testing -- LGTM! I even like the new, bare sliders. I don't _think_ that the lack of horizontal lines on them make it any less obvious that the user can interact with them and slide them around. Hopefully that's implied enough by the hollow gray space they rest in.

**Pull request merged**